apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v2/eobByPatientIdAll.json

```diff
@@ -2730,7 +2730,7 @@
         },
         "productOrService" : {
           "coding" : [ {
-            "system" : "https://bluebutton.cms.gov/resources/variables/hcpcs_cd",
+            "system" : "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
```

Comment:
Isn't this a breaking change?

Reply:
Confirmed: this is a breaking change. We do **not** want to do that, as things stand right now.

Instead, we'll need to _add_ a `Coding` to the `CodeableConcept` that has the correct `system`, and leave the one with the goofy `system` there for backwards-compatibility purposes. In JSON terms, you're adding another `system+code` dict to the `coding` array.

This will require changes to our current SAMHSA filtering to work correctly. Right now, I think the relevant portion of the code is this:

```java
  private boolean containsSamhsaProcedureCode(CodeableConcept procedureConcept) {
    for (Coding procedureCoding : procedureConcept.getCoding()) {
      if (TransformerConstants.CODING_SYSTEM_HCPCS.equals(procedureCoding.getSystem())) {
        if (isSamhsaCptCode(procedureCoding)) return true;
      } else {
        /*
         * Fail safe: if we don't know the procedure Coding system, assume the code is
         * SAMHSA.
         */
        return true;
      }
    }

    // No blacklisted procedure Codings found: this procedure isn't SAMHSA-related.
    return false;
  }
```

In pseudo-code terms, we'll instead need it to look more like this:

```java
  private boolean containsSamhsaProcedureCode(CodeableConcept procedureConcept) {
    // Does the CodeableConcept have a legit HCPCS Coding?
    Optional<Coding> hcpcsCoding = findHcpcsCoding(procedureConcept);

    // Check that Coding to see if it's blacklisted.
    if(hcpcsCoding.isPresent() && isSamhsaCptCode(procedureCoding)) {
      return true;
    } else if(hcpcsCoding.isPresent() && hasUnknownSystem(procedureConcept)) {
        /*
         * Fail safe: if we don't know the procedure Coding system, assume the code is
         * SAMHSA.
         */
        return true;
    } else {
      // Otherwise, this only has known & non-SAMHSA-blacklisted procedure codes.
      return false;
    }
  }

  private boolean findHcpcsCoding(CodeableConcept procedureConcept) {
    ...
  }

  private boolean hasUnknownSystem(CodeableConcept procedureConcept) {
    Set<String> codingSystems = procedureConcept.getCoding().stream().map(coding -> coding.getSystem()).collect(Collectors.toSet());

    if (codingSystem.size() == 2 && codingSystem.contains(TransformerConstants.CODING_SYSTEM_HCPCS) && codingSystem.contains(CcwCodebookVariable.HCPCS_CD.whatever())) {
      /* add a comment here explaining why this complex check is needed and referencing the JIRA ticket */
      return false;
    } else if (codingSystem.size() == 1 && codingSystem.contains(TransformerConstants.CODING_SYSTEM_HCPCS)) {
      return false;
    } else {
      return true;
    }
  }
```

Reply:
@karlmdavis We ended up needing to deviate from this some so would appreciate the extra scrutiny on what was implemented vs what is here.

Reply:
This was addressed.